main.nf

```diff
@@ -214,35 +201,35 @@ process bwa {
 
     module 'bioinfo-tools'
     module 'bwa'
-    module 'bamtools'
     module 'samtools'
     module 'BEDTools'
-
+    
     cpus 2
-    memory '16 GB'
-    time '120h'
-
-    publishDir "$results_path/bwa"
-
+    memory { 16.GB * task.attempt }
+    time { 120.h * task.attempt }
+    errorStrategy { task.exitStatus == 143 ? 'retry' : 'terminate' }
+    maxRetries 3
+    maxErrors '-1'
+    
+    publishDir "${params.outdir}/bwa", mode: 'copy'
+    
     input:
     file (reads:'*') from trimmed_reads
     set val(prefix) from name_for_bwa
 
     output:
-    file '*.sort.bam' into bam_picard
-    file '*.sort.bam.bai' into results
-    file '*.sort.bed' into results
-
+    file '*.sorted.bam' into bam_picard
+    file '*.sorted.bam.bai'
+    file '*.sorted.bed'
+    stdout into bwa_logs
+    
+    script:
     """
-    bwa mem -M $genomebwa ${reads} > ${prefix}.sam
-    samtools view -bT $genomebwa ${prefix}.sam > ${prefix}.bam
-    samtools sort ${prefix}.bam ${prefix}.sort
-    samtools index ${prefix}.sort.bam
-    rm ${prefix}.sam
-    rm ${prefix}.bam
-    bedtools bamtobed -i ${prefix}.sort.bam | sort -k 1,1 -k 2,2n -k 3,3n -k 6,6 > ${prefix}.sort.bed
+    bwa mem -M $index $reads | samtools view -b > ${prefix}.bam
```

Comment:
Hi, the -bT $genomebwa can be removed.
But I will test this part. Thanks!
/chuan

On Thu, Jun 16, 2016 at 11:47 AM, Phil Ewels <notifications@github.com>
wrote:

> In main.nf
> <https://github.com/SciLifeLab/NGI-ChIPseq/pull/3#discussion_r67316731>:
>
> >      """
> > -    bwa mem -M $genomebwa ${reads} > ${prefix}.sam
> > -    samtools view -bT $genomebwa ${prefix}.sam > ${prefix}.bam
> > -    samtools sort ${prefix}.bam ${prefix}.sort
> > -    samtools index ${prefix}.sort.bam
> > -    rm ${prefix}.sam
> > -    rm ${prefix}.bam
> > -    bedtools bamtobed -i ${prefix}.sort.bam | sort -k 1,1 -k 2,2n -k 3,3n -k 6,6 > ${prefix}.sort.bed
> > +    bwa mem -M $index $reads | samtools view -b > ${prefix}.bam
>
> Note @chuan-wang <https://github.com/chuan-wang> - important change here
> that needs testing.
>
> Instead of creating a temporary .sam file and converting it to .bam, I'm
> piping the output to samtools to convert to .bam directly. This should be
> faster and save filespace.
>
> *However*, one but I was confused about was why you were specifying -T
> $genomebwa in the conversion step. I don't usually do this, I think it's
> only needed when some SAM header info is missing, or other stuff. But need
> to be sure that it still works as I have it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/SciLifeLab/NGI-ChIPseq/pull/3/files/a3d3182481382cee2201bde58e3c0b9567daab61#r67316731>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ARdKNKXIdKIGeKbjF9vvKQu4gLqwN0jxks5qMRubgaJpZM4I3LyR>
> .
>